File: Movie/Models/Person.swift
//
//  Person.swift
//  Movie
//
//

import Foundation

struct Person: Codable, Identifiable  {
    let id: Int
    let name: String
    let placeOfBirth: String
    let biography: String
    let birthday: Date
    let profilePath: String?

    
    var profilePhotoURL: URL {
        return URL(string: "https://image.tmdb.org/t/p/original\(profilePath ?? "")")!
    }
    
    enum CodingKeys: String, CodingKey {
        case id
        case name
        case placeOfBirth = "place_of_birth"
        case biography
        case birthday
        case profilePath = "profile_path"
    }
}

File: shared/src/commonMain/kotlin/com/mehmetaliocak/movie_app/models/Person.kt
package com.mehmetaliocak.movie_app.models

import com.mehmetaliocak.movie_app.util.asPhotoUrl
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Person(
    val id: Long,
    val name: String,
    @SerialName("place_of_birth")
    val placeOfBirth: String?,
    val biography: String,
    val birthday: String?,
    @SerialName("profile_path")
    val profilePath: String?
) {

    fun profilePhotoURL(): String {
        return profilePath.asPhotoUrl()
    }
}
